src/rust/lib/graph-editor/src/component/node/port.rs

```diff
@@ -0,0 +1,478 @@
+            inner_radius : Cell::new(spec.inner_radius),
+            position     : Cell::new(spec.position),
+            direction    : Cell::new(spec.direction),
+            view
+        });
+        Self {data} . init()
+    }
+
+    fn init(self) -> Self {
+        self.init_shape();
+        self.update_sprite_position();
+        self
+    }
+
+    /// Sets the port's position.
+    pub fn set_position(&self, position:Angle<Degrees>) {
+        self.data.position.set(position);
+        self.update_sprite_position();
+    }
+
+    /// Sets the ports inner radius.
+    pub fn set_inner_radius(&self, inner_radius:f32) {
+        self.data.inner_radius.set(inner_radius);
+        if let Some(t) = self.data.view.data.borrow().as_ref() {
+            t.shape.inner_radius.set(inner_radius);
+        }
+        self.update_sprite_position()
+    }
+
+    /// Sets the ports height.
+    pub fn set_height(&self, height:f32) {
+        self.data.height.set(height);
+        if let Some(t) = self.data.view.data.borrow().as_ref() {
+            t.shape.height.set(height);
+        }
+    }
+
+    /// Sets the ports width.
+    pub fn set_width(&self, width:Angle<Degrees>) {
+        self.data.width.set(width);
+        if let Some(t) = self.data.view.data.borrow().as_ref() {
+            t.shape.width.set(width.value.to_radians());
+        }
+    }
+
+    /// Update the position of the sprite according to the Port specification.
+    /// The position is given along a circle, thus the position and rotation of the sprite
+    /// are tied together, so the Port always point in the right direction.
+    /// Needs to be called whenever the `position` or `inner_radius` is modified.
+    fn update_sprite_position(&self) {
+        let inner_radius = self.data.inner_radius.get();
+        let position     = self.data.position.get();
+
+        let translation_vector = Vector3::new(0.0,inner_radius,0.0);
+        let rotation_vector    = -Vector3::new(0.0,0.0,position.rad().value);
+        let rotation           = nalgebra::Rotation3::new(rotation_vector);
+        let translation        = rotation * translation_vector;
+        self.data.view.display_object.set_position(translation);
+        self.data.view.display_object.set_rotation(rotation_vector);
+    }
+
+    /// Update the shape parameters with values from our `Specification`.
+    /// Should only be used upon initialisation.
+    fn init_shape(&self) {
+        if let Some(t) = self.data.view.data.borrow().as_ref() {
+            let height       = self.data.height.get();
+            let inner_radius = self.data.inner_radius.get();
+            let width        = self.data.width.get();
+            let direction    = self.data.direction.get();
+
+            t.shape.update_parameters(height,inner_radius,width,direction);
+        }
+    }
+}
+
+impl<T:PortShapeViewDefinition> Default for Port<T> {
+    fn default() -> Self {
+        Self::new(Specification::default())
+    }
+}
```

Comment:
derive Default for Port

Reply:
`Port` requires some initialisation logic (e.g, setting the initial position of the sprite), so there needs to be a call to `new` which requires the manual implementation.

Reply:
then do not name it `new`. `new` should be equal to `default`. Name it `new_with_something` or osmething